src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -49,41 +57,142 @@ module.exports = function<T, P, I, C>(config : HostConfig<T, P, I, C>) {
   function updateFunctionalComponent(current, workInProgress) {
     var fn = workInProgress.type;
     var props = workInProgress.pendingProps;
-    console.log('update fn:', fn.name);
     var nextChildren = fn(props);
     reconcileChildren(current, workInProgress, nextChildren);
     workInProgress.pendingWorkPriority = NoWork;
   }
 
-  function updateHostComponent(current, workInProgress) {
-    console.log(
-      'host component', workInProgress.type, 
-      typeof workInProgress.pendingProps.children === 'string' ? workInProgress.pendingProps.children : ''
-    );
+  function scheduleUpdate(fiber: Fiber, pendingState: PendingState, priorityLevel : PriorityLevel): void {
+    const { scheduleLowPriWork } = getScheduler();
+    fiber.pendingState = pendingState;
+    while (true) {
+      if (fiber.pendingWorkPriority === NoWork ||
+          fiber.pendingWorkPriority >= priorityLevel) {
+        fiber.pendingWorkPriority = priorityLevel;
+      }
+      // Duck type root
+      if (fiber.stateNode && fiber.stateNode.containerInfo) {
+        const root : FiberRoot = (fiber.stateNode : any);
+        scheduleLowPriWork(root, priorityLevel);
+        return;
+      }
+      if (!fiber.return) {
+        throw new Error('No root!');
+      }
+      fiber = fiber.return;
+    }
+  }
+
+  // Class component state updater
+  const updater = {
+    enqueueSetState(instance, partialState) {
+      const fiber = instance._fiber;
+      let pendingState = fiber.pendingState;
+
+      // Append to pending state queue
+      const nextPendingStateNode = createPendingState(partialState);
+      if (pendingState === null) {
+        pendingState = nextPendingStateNode;
+      } else {
+        if (pendingState.tail === null) {
+          pendingState.next = nextPendingStateNode;
+        } else {
+          pendingState.tail.next = nextPendingStateNode;
+        }
+        pendingState.tail = nextPendingStateNode;
+      }
+
+      // Must schedule an update on both alternates, because we don't know tree
+      // is current.
+      scheduleUpdate(fiber, pendingState, LowPriority);
+      if (fiber.alternate) {
+        scheduleUpdate(fiber.alternate, pendingState, LowPriority);
```

Comment:
Ooooh, okay. I think I get it now! Thanks for the detailed explanation (and for being patient as I try to keep up, haha).